core/whistle_services-1.0.0/src/bookkeepers/wh_bookkeeper_braintree.erl

```diff
@@ -252,27 +252,31 @@ commit_transactions(BillingId, _Transactions, _Try) ->
 %%
 %% @end
 %%--------------------------------------------------------------------
--spec charge_transactions(ne_binary(), wh_json:objects()) ->
-                                 wh_transactions:wh_transactions().
--spec charge_transactions(ne_binary(), wh_json:objects(), dict()) ->
-                                 wh_transactions:wh_transactions().
+-spec charge_transactions(ne_binary(), wh_json:objects()) -> wh_json:objects().
+-spec charge_transactions(ne_binary(), wh_json:objects(), dict()) -> wh_json:objects().
 charge_transactions(BillingId, Transactions) ->
     charge_transactions(BillingId, Transactions, dict:new()).
 
 charge_transactions(BillingId, [], Dict) ->
     dict:fold(
-      fun(Code, JObjs, Acc) when Code =:= ?CODE_TOPUP ->
-              handle_topup(JObjs, Acc, BillingId);
-         (Code, JObjs, Acc) ->
-              handle_charged_transactions(Code, JObjs, Acc, BillingId)
-      end
-      ,[]
-      ,Dict
-     );
+        fun(Code, JObjs, Acc) when Code =:= ?CODE_TOPUP ->
+            case handle_topup(BillingId, JObjs) of
+                'true' -> Acc;
+                'false' -> JObjs ++ Acc
+            end;
+        (Code, JObjs, Acc) ->
+             case handle_charged_transactions(BillingId, Code, JObjs) of
+                'true' -> Acc;
+                'false' -> JObjs ++ Acc
+            end
+        end
+        ,[]
+        ,Dict
+    );
 charge_transactions(BillingId, [Transaction|Transactions], Dict) ->
-    Code = wh_json:get_value(<<"code">>, Transaction),
+    Code = wh_json:get_value(<<"pvt_code">>, Transaction),
     L = case dict:find(Code, Dict) of
```

Comment:
Can you just do `L = dict:append(Code, Transaction, Dict)` instead of a case clause?